### ISSUE TITLE
[help] Hard cropping?

### ISSUE THREAD

**Issue by Shohreh** [pdfcpu/pdfcpu#620]
Hello,

This is not a bug report, but I didn't where else to ask for help.

Looks like I'm doing it wrong: I just tried pdfcpu to perform a hard crop so that the original info is really gone from the output PDF, but for some reason it's still there when I open it in LibreOffice Draw.

I need to remove the header on most of the pages, so tried the following:

`pdfcpu.exe crop -p 19-24,26-97 -- "[0 0 424 600]" input.pdf output.pdf`

What's the right command?

Thank you.

![image](https://github.com/pdfcpu/pdfcpu/assets/1541861/593ff04a-f504-4276-9a19-3b9f01234a02)

**Comment by hhrutter** [comment]
The right place for questions is => https://github.com/pdfcpu/pdfcpu/discussions

The crop box takes care of general page content clipping.
I don't see why this should not be working in your case as long as you are aware of your pages boxes:

To get it working first pick one page and analyze its page boundaries, let's say 1:

`pdfcpu boxes -p 1 list`

you should see smth like:

```
Page 1: rot=+0 orientation:portrait
  MediaBox (points) (0.00, 0.00, 595.00, 842.00) w=595.00 h=842.00 ar=0.71 (inherited)
   CropBox (points) (0.00, 0.00, 595.00, 842.00) w=595.00 h=842.00 ar=0.71 (default)
   TrimBox (points) (0.00, 0.00, 595.00, 842.00) w=595.00 h=842.00 ar=0.71 (default)
  BleedBox (points) (0.00, 0.00, 595.00, 842.00) w=595.00 h=842.00 ar=0.71 (default)
    ArtBox (points) (0.00, 0.00, 595.00, 842.00) w=595.00 h=842.00 ar=0.71 (default)
```

**Comment by Shohreh** [comment]
Thanks. Sorry about the wrong section.

Here's what it says. I need to really crop/trim the header in most of the pages so that the data is totally gone in the output file
```
pdfcpu.exe boxes list  -p 20 input.pdf
listing mediaBox, cropBox, trimBox, bleedBox, artBox for input.pdf
pages: 20
Page 20: rot=+0 orientation:portrait
  MediaBox (points) (0.00, 0.00, 424.80, 640.80) w=424.80 h=640.80 ar=0.66
   CropBox (points) (0.00, 39.92, 424.15, 640.80) w=424.15 h=600.88 ar=0.71
   TrimBox (points) (0.00, 39.92, 424.15, 640.80) w=424.15 h=600.88 ar=0.71 (default)
  BleedBox (points) (0.00, 39.92, 424.15, 640.80) w=424.15 h=600.88 ar=0.71 (default)
    ArtBox (points) (0.00, 39.92, 424.15, 640.80) w=424.15 h=600.88 ar=0.71 (default)
```
![image](https://github.com/pdfcpu/pdfcpu/assets/1541861/6ea644b5-a35e-4436-85a2-6a14bebf3227)

**Comment by clysss** [comment]
the @Shohreh Shohreh question is often a issue when you have go to a printer with a cropped PDF as cropped data are making them ##### !
to "hard crop", you have to change mediabox size : pdfcpu can do that ...
for example 
abs coord : 
pdfcpu box add -u mm -- "media:[5 5 224 307]" in.pdf out.pdf
relat (marges) size :
pdfcpu box add -u mm -- "media:5 5 5 5" in.pdf out.pdf
and after, you can also remove the "other" boxes :
pdfcpu box remove -u mm -- "c, t, b, a" "out.pdf"
if you need to remove cropBox, trimBox, bleedBox, artBox for out.pdf

Still 2 things that pdfcpu box seems not to be able do :
1.  crop command permit to create a centered box with command like : 
pdfcpu crop -u mm -- "pos:c, off:0 0, dim: 214 301 abs" in.pdf out.pdf 
but 
pdfcpu box add -u mm -- "media:pos:c, off:0 0, dim: 214 301 abs" in.pdf xout.pdf
don't work (I imagine this is because pos is relative to media...)
2. align mediabox to cropbox (or trimbox)
most "visual" software permit to modify cropbox. After, you need to fixup this during preflight 
cf https://superuser.com/questions/127568/how-to-crop-permanently-in-acrobat
to Set page geometry boxes : MediaBox to TrimBox -or cropbox-
so something that we should write like :
pdfcpu box add -- "media:crop" in.pdf out.pdf
but it doesn't work :(( 
This 2. point should be very usefull to solve also @Shohreh issue and also permit not to implement the point 1. as we could just use the crop command first and then the box media:crop ..

**Comment by Shohreh** [comment]
Thanks for the infos. I found a solution later based in Python, by setting and deleting "redaction annotation", but pdfcpu's "remove" command is good to know.

```
import fitz
doc = fitz.open(item)
WIDTH = doc[0].mediabox.width
HEIGHT = doc[0].mediabox.height
rect_header = fitz.Rect(0,0,WIDTH,53)
rect_footer = fitz.Rect(0,770,WIDTH,790)
page.add_redact_annot(rect_header)
page.add_redact_annot(rect_footer)
page.apply_redactions()
doc.save("redacted.pdf")
```

**Comment by clysss** [comment]
Yes I found your thread somewhere else;) searching also my way....

**Comment by clysss** [comment]
@hhrutter : hello, could you tag this thread as Feature request  (?
Thx

#Still 2 things that pdfcpu box seems not to be able do :

    crop command permit to create a centered box with command like :
    pdfcpu crop -u mm -- "pos:c, off:0 0, dim: 214 301 abs" in.pdf out.pdf
    but
    pdfcpu box add -u mm -- "media:pos:c, off:0 0, dim: 214 301 abs" in.pdf xout.pdf
    don't work (I imagine this is because pos is relative to media...)
    align mediabox to cropbox (or trimbox)
    most "visual" software permit to modify cropbox. After, you need to fixup this during preflight
    cf https://superuser.com/questions/127568/how-to-crop-permanently-in-acrobat
    to Set page geometry boxes : MediaBox to TrimBox -or cropbox-
    so something that we should write like :
    pdfcpu box add -- "media:crop" in.pdf out.pdf
    but it doesn't work :((
    This 2. point should be very usefull to solve also @Shohreh issue and also permit not to implement the point 1. as we could just use the crop command first and then the box media:crop .
